### PR TITLE
Add transitive GetComponent methods

### LINKED ISSUE #28: Provide Get…Component helpers for transitive dependencies.
For example, the `AcidTubeComponent` should be able to use `GetHealthComponent().Alive()` instead of `GetAlienBuildableComponent().GetBuildableComponent().GetHealthComponent().Alive()`.

Apart from being more readable, this also allows the use of the current `REGISTER_THINKER` macro for entitites that don't directly depend on the `ThinkingComponent`.

### PR DESCRIPTION
**Pull request by Kangz**
Fixes #28 

PTAL, it also adds const versions for Get* that were missing.

### CODE REVIEW

File: CBSE.py [hunk 1]
```diff
@@ -197,6 +217,10 @@ def get_own_required_component_names(self):
         #TODO
         return ['r_' + c.get_type_name() for c in self.requiredComponents]
 
+    def get_own_further_dependencies(self):
```
Comment (Viech) [inline CBSE.py]: Not really related to this PR, but do we even use any of the `own`-labeled functions?

File: templates/Backend.h [hunk 2]
```diff
@@ -237,11 +237,20 @@ class AllComponents {
...
 			{% endfor %}
+
+			{% for (dependency, firstLevel) in component.get_own_further_dependencies().items() %}
+				/**
+				 * @return A reference to the {{dependency.get_type_name()}} of the owning entity.
+				 */
+				{{dependency.get_type_name()}}& Get{{dependency.get_type_name()}}();
+				const {{dependency.get_type_name()}}& Get{{dependency.get_type_name()}}() const;
+
+			{% endfor %}
```
Comment (Viech) [inline templates/Backend.h]: NIT: This adds an extra newline between the last `Get` declaration and the one of `entity`. :wink:

**Comment by Viech** [comment]
Using this commit with Unvanquished, I do get compile errors:

```
In file included from /home/eris/src/Unvanquished/src/sgame/backend/../components/DeferredFreeingComponent.h:28:0,
                 from /home/eris/src/Unvanquished/src/sgame/backend/CBSEComponents.h:6,
                 from /home/eris/src/Unvanquished/src/sgame/backend/CBSEEntities.h:11,
                 from /home/eris/src/Unvanquished/src/sgame/CBSE.h:24,
                 from /home/eris/src/Unvanquished/src/sgame/Beacon.cpp:28:
/home/eris/src/Unvanquished/src/sgame/backend/../components/../backend/CBSEBackend.h:721:18: error: 'TeamComponent& AlienClassComponentBase::GetTeamComponent()' cannot be overloaded
   TeamComponent& GetTeamComponent();
                  ^~~~~~~~~~~~~~~~
/home/eris/src/Unvanquished/src/sgame/backend/../components/../backend/CBSEBackend.h:696:18: error: with 'TeamComponent& AlienClassComponentBase::GetTeamComponent()'
   TeamComponent& GetTeamComponent();
                  ^~~~~~~~~~~~~~~~
/home/eris/src/Unvanquished/src/sgame/backend/../components/../backend/CBSEBackend.h:722:24: error: 'const TeamComponent& AlienClassComponentBase::GetTeamComponent() const' cannot be overloaded
   const TeamComponent& GetTeamComponent() const;
                        ^~~~~~~~~~~~~~~~
/home/eris/src/Unvanquished/src/sgame/backend/../components/../backend/CBSEBackend.h:697:24: error: with 'const TeamComponent& AlienClassComponentBase::GetTeamComponent() const'
   const TeamComponent& GetTeamComponent() const;
                        ^~~~~~~~~~~~~~~~
```

The output continues "like this" with lots of errors, all of them seem to be related to the `TeamComponent` though. I'd propose to checkout the PRs branch within unvanquished if you want to get more details.

Here's an excert from `entities.yaml`, manually stripped down to components and entities that involve the `TeamComponent` directly (though not double checked if these are really all occurences):

```
components:
    Team:
        parameters:
            team: team_t
    Client:
        parameters:
            clientData: gclient_t*
        requires:
            Team:
    AlienClass:
        requires:
            Client:
            Team:
                team: !!plain TEAM_ALIENS
            Armor:
            Knockback:
            Health:
    HumanClass:
        requires:
            Client:
            Team:
                team: !!plain TEAM_HUMANS
            Armor:
            Knockback:
            Health:
    Buildable:
        messages:
            - PrepareNetCode
            - Die
        requires:
            Health:
            Thinking:
            Team:
    AlienBuildable:
        messages:
            - Damage
            - Die
        requires:
            Buildable:
            Team:
                team: !!plain TEAM_ALIENS
            Ignitable:
    HumanBuildable:
        messages:
            - Die
        requires:
            Buildable:
            Team:
                team: !!plain TEAM_HUMANS
    Spawner:
        messages:
            - Die
        requires:
            Team:
            Thinking:
entities:
    Client:
        components:
            Client:
            Team:
                team: !!plain TEAM_NONE
```

File: CBSE.py (hunk 1 above)
Comment (Kangz) [inline CBSE.py]: The own labels are for inheritance that's not implemented yet. Basically get_messages() would return all the messages handled by this component while get_own_messages() would return all the messages handled by this component that are not handled by the component's parent components.

It is very confusing and I'm thinking of dropping it to simplify the code, cleanup stuff, then reimplement inheritance from scratch.

**Comment by Kangz** [comment]
Thanks for the minimal repro case, I will TAL!

File: templates/Backend.h (hunk 2 above)
Comment (Kangz) [inline templates/Backend.h]: Done

**Comment by Kangz** [comment]
Uploaded a fixed version, thanks for the review!

**Comment by Viech** [comment]
Tested in Unvanquished, works just fine. Thanks!

**Pull request merged**